### ISSUE TITLE
bug: No error is returned from lightpush if a peer is full

### ISSUE THREAD

**Issue by chair28980** [waku-org/nwaku#2335]
## Problem

I ran into a problem that I had a peer that had a full disk and `lightPush.send` returned no error, but the data returned from `store.queryGenerator` was empty.

More specifically there were no `errors` in `SendResult`, and the peer was in `recipients`.

## Proposed Solutions

It would be better to return either `REMOTE_PEER_FAULT` or `REMOTE_PEER_REJECTED` if the peer is full and cannot store data. I haven't checked the code, but most likely this issue affects not only `js-waku` but also `nwaku` (and possibly `go-waku` as well).

## Notes

- js-waku version: `"@waku/sdk": "^0.0.20"`
- Environment (browser software and version, system software and version): Firefox & Chromium on Linux
- Nodes connected to: nwaku 0.21.1
---
Originally reported in https://github.com/waku-org/js-waku/issues/1715 by https://github.com/agazso

**Comment by NagyZoltanPeter** [comment]
@chair28980: I reviewed this bug description, but I don't think its a bug or at least we do not have such expectation from lightpush to know if a store service received and was able to store the message. Although we have such kind of ideas in the stack for the future. But apart from that there is nothing to fix in the current implementation about it.
I don't think it is even can be part of store v3 and sync efforts as it is more a client side duty to check. Lightpush can answer that the message is relayed to the network or not, what happens afterward is out of scope and interest of the lightpush service itself.

@jm-clius: WDYT about this in the light of upcoming lightpush enhancement efforts?

**Comment by jm-clius** [comment]
Indeed. For now the only "requested service" implied in lightpush is to request the service node to publish the message to the relay network. If this target node could not relay the message (as far as it could determine), it should return an error. This message may or may not be stored in the network by any number of store nodes, some of which may be full. The fact that the target service node also happens to be a store node is, from lightpush pov, just a coincidence and the success of the store operation should not affect lightpush service as it stands now.

In future we may explicitly request the target node to store a message via lightpush, in which case it should indeed return an error on failure.

TL;DR: I'd suggest closing this issue as not a bug.
